File: app/src/main/java/de/blankedv/lanbahnpanel/SensorElement.java
package de.blankedv.lanbahnpanel;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.util.Log;
import static de.blankedv.lanbahnpanel.LanbahnPanelApplication.*;
import static de.blankedv.lanbahnpanel.LinePaints.*;


public class SensorElement extends ActivePanelElement {
	
	public SensorElement(int x, int y, String name, int adr) {
		super(x, y, name,  adr);
        
	}
	
	public SensorElement() {
		super();
	}

	@Override
	public void doDraw(Canvas canvas) {

		if (x2 != INVALID_INT) {  // draw dashed line as sensor
			// read data from SX bus and set red/gray dashed line accordingly

			if (state == STATE_FREE) {
				
					canvas.drawLine(x * prescale, y * prescale, x2 * prescale, y2
							* prescale, linePaintGrayDash);

			} else if (state == STATE_OCCUPIED) {
				
					canvas.drawLine(x * prescale, y * prescale, x2 * prescale, y2
							* prescale,  linePaintRedDash);
	

			}  else if (state == STATE_INROUTE) {
				
				    canvas.drawLine(x * prescale, y * prescale, x2 * prescale, y2
						* prescale, linePaintDarkYellowDash);
				

			}
			else if (state == STATE_UNKNOWN) {
				
					canvas.drawLine(x * prescale, y * prescale, x2 * prescale, y2
							* prescale, linePaintGrayDash);
			}
		} else {
			// draw lamp type of sensor   s_on.png etc

			int h, w;
			Bitmap bm;

			StringBuilder bmName = new StringBuilder("sensor");

			if ( (state == STATE_FREE)  || (state == STATE_UNKNOWN)){
				bmName.append("_off");
			} else {
				bmName.append("_on");
			}

			bm = bitmaps.get(bmName.toString());
			if (bm == null) {
                Log.e(TAG,
                        "error, bitmap not found with name="
                                + bmName.toString());
            } else {
                h = bm.getHeight() / 2;
                w = bm.getWidth() / 2;
                canvas.drawBitmap(bm, x * prescale - w, y * prescale - h, null); // center
                // bitmap
            }
		}
		if (drawAddresses2)
			doDrawAddresses(canvas);
	}
	

}

File: app/src/main/java/de/blankedv/lanbahnpanel/SensorElement.kt
package de.blankedv.lanbahnpanel

import android.graphics.Bitmap
import android.graphics.Canvas
import android.util.Log
import de.blankedv.lanbahnpanel.LanbahnPanelApplication.*
import de.blankedv.lanbahnpanel.LinePaints.*


class SensorElement : ActivePanelElement {

    constructor(x: Int, y: Int, name: String, adr: Int) : super(x, y, name, adr) {

    }

    constructor() : super() {}

    override fun doDraw(canvas: Canvas) {

        if (x2 != INVALID_INT) {  // draw dashed line as sensor
            // read data from SX bus and set red/gray dashed line accordingly

            if (state == ActivePanelElement.STATE_FREE) {

                canvas.drawLine((x * prescale).toFloat(), (y * prescale).toFloat(), (x2 * prescale).toFloat(), (y2 * prescale).toFloat(), INSTANCE.getLinePaintGrayDash())

            } else if (state == ActivePanelElement.STATE_OCCUPIED) {

                canvas.drawLine((x * prescale).toFloat(), (y * prescale).toFloat(), (x2 * prescale).toFloat(), (y2 * prescale).toFloat(), INSTANCE.getLinePaintRedDash())


            } else if (state == ActivePanelElement.STATE_INROUTE) {

                canvas.drawLine((x * prescale).toFloat(), (y * prescale).toFloat(), (x2 * prescale).toFloat(), (y2 * prescale).toFloat(), INSTANCE.getLinePaintDarkYellowDash())


            } else if (state == ActivePanelElement.STATE_UNKNOWN) {

                canvas.drawLine((x * prescale).toFloat(), (y * prescale).toFloat(), (x2 * prescale).toFloat(), (y2 * prescale).toFloat(), INSTANCE.getLinePaintGrayDash())
            }
        } else {
            // draw lamp type of sensor   s_on.png etc

            val h: Int
            val w: Int
            val bm: Bitmap?

            val bmName = StringBuilder("sensor")

            if (state == ActivePanelElement.STATE_FREE || state == ActivePanelElement.STATE_UNKNOWN) {
                bmName.append("_off")
            } else {
                bmName.append("_on")
            }

            bm = bitmaps[bmName.toString()]
            if (bm == null) {
                Log.e(TAG,
                        "error, bitmap not found with name=" + bmName.toString())
            } else {
                h = bm.height / 2
                w = bm.width / 2
                canvas.drawBitmap(bm, (x * prescale - w).toFloat(), (y * prescale - h).toFloat(), null) // center
                // bitmap
            }
        }
        if (drawAddresses2)
            doDrawAddresses(canvas)
    }


}
